lib/vmpooler/api/v1.rb

```diff
@@ -173,6 +173,72 @@ class V1 < Sinatra::Base
       JSON.pretty_generate(result)
     end
 
+    get "#{api_prefix}/token/:token/?" do
+      content_type :json
+
+      result = {}
+
+      Vmpooler::API.settings.config[:auth] ? status(401) : status(404)
```

Comment:
Regardless the client receives a `40x` response, yes? How does it continue to the remainder of this route?